nfc/NfcCx/Cx/NfcCxSC.cpp

```diff
@@ -2834,6 +2874,98 @@ Return Value:
     return fLoadKey;
 }
 
+BOOL NfcCxSCInterfaceValidateSwitchProtocolCommand(
+    _In_bytecount_(InputBufferLength) PBYTE InputBuffer,
+    _In_ DWORD InputBufferLength
+)
+{
+    BOOL retval = FALSE;
+
+    TRACE_FUNCTION_ENTRY(LEVEL_VERBOSE);
+
+    TRACE_LINE(LEVEL_INFO, "ValidateSwitchProtocolCommand Size = %x\n", InputBufferLength);
+    if (InputBufferLength < PCSC_SWITCH_PROTOCOL_APDU_SIZE)
+    {
+        TRACE_LINE(LEVEL_ERROR, "Invalid APDU buffer size");
+        goto Done;
+    }
+    else
+    {
+        PPcscCommandApduInfo cmdApdu = (PPcscCommandApduInfo)InputBuffer;
+        if ((0xFF == cmdApdu->Cla) &&
+            (PcscEnvelopeCmd == cmdApdu->Ins) &&
+            (0x00 == cmdApdu->P1) && ((0x02 == cmdApdu->P2)))
+        {
```

Comment:
Any magic number used in code should be #defined or have comment explaining why this constant was used.